apps sc: upgrade to upstream tekton chart 1.1.0

Ajarmar opened the pull request:
<!-- Choose your PR title carefully as it will be used as the entry in the changelog! -->

<!-- markdownlint-disable MD041 -->
> [!warning]
> **This is a public repository, ensure not to disclose:**
>
> - [ ] personal data beyond what is necessary for interacting with this pull request, nor
> - [ ] business confidential information, such as customer names.

### What kind of PR is this?

**Required**: Mark one of the following that is applicable:

- [ ] kind/feature       <!-- This PR adds a new feature -->
- [x] kind/improvement   <!-- This PR changes an existing feature -->
- [ ] kind/deprecation   <!-- This PR removes an existing feature -->
- [ ] kind/documentation <!-- This PR contains documentation -->
- [ ] kind/clean-up      <!-- This PR cleans up technical debt -->
- [ ] kind/bug           <!-- This PR fixes a bug -->
- [ ] kind/other         <!-- This PR does something else -->

_Optional_: Mark one or more of the following that are applicable:

> [!important]
> Breaking changes should be marked `kind/admin-change` or `kind/dev-change` depending on type
> Critical security fixes should be marked with `kind/security`

- [ ] kind/admin-change   <!-- This PR introduces an admin facing change, add "Platform Administrator notice" section -->
- [ ] kind/dev-change     <!-- This PR introduces a dev facing change, add "Application Developer notice" section -->
- [ ] kind/security       <!-- This PR introduces a critical security fix, add "Security notice" section -->
- [ ] [kind/adr](set-me\) <!-- This PR implements an ADR, add the link -->

<!-- Uncomment the additional sections that applies. -->

<!-- Additional information to be added in the release notes
### Release notes
...
-->

<!-- Additional information with kind/admin-change
### Platform Administrator notice
...
-->

<!-- Add additional information with kind/dev-change
### Application Developer notice
...
-->

<!-- Add additional information with kind/security
### Security notice
...
-->

### What does this PR do / why do we need this PR?

<!-- Add description of the change -->
Upgrade to Tekton 1.1.0. We previously used our own chart for this, but there is now an [upstream chart](https://github.com/cdfoundation/tekton-helm-chart) that seems to be frequently updated, so I switched to that.

<!-- Add all issues that are fixed by this PR, use "Part of" instead of "Fixes" if you want to keep issues open. -->
- Fixes https://github.com/elastisys/ck8s-issue-tracker/issues/559

#### Information to reviewers

<!--
Any additional information reviews should know.

How to run / how to test.

Include screenshots if applicable to help explain these changes.
--->

There is a migration script, mostly to handle CRDs.

We previously had a ServiceMonitor, but there isn't one in the upstream chart. I was trying to make it work without one but I'm not sure it's possible [with the prometheus operator](https://github.com/prometheus-community/helm-charts/tree/3a96cf8/charts/kube-prometheus-stack#prometheusioscrape). I can add a ServiceMonitor in its own chart, but I wanted to get the PR up first so people can start looking at it.

#### Checklist

<!-- This section is not added to the changelog or release notes, it is to help you as a contributor and reviewers. -->

- [x] Proper commit message prefix on all commits
    <!-- Example of commit message prefixes:
    - all: changes to multiple areas
    - apps: changes to applications running in all clusters
    - apps sc: changes to applications running in service clusters
    - apps wc: changes to applications running in workload clusters
    - bin: changes to management binaries
    - config: changes to configuration
    - docs: changes to documentation
    - release: release related
    - scripts: changes to scripts
    - tests: changes to tests
    --->
- Change checks:
    - [ ] The change is transparent
    - [ ] The change is disruptive
    - [ ] The change requires no migration steps
    - [x] The change requires migration steps
    - [x] The change updates CRDs
    - [ ] The change updates the config _and_ the schema
- Documentation checks:
    - [x] The [public documentation](https://github.com/elastisys/welkin) required no updates
    - [ ] The [public documentation](https://github.com/elastisys/welkin) required an update - [link to change](set-me\)
- Metrics checks:
    - [ ] The metrics are still exposed and present in Grafana after the change
    - [ ] The metrics names didn't change (Grafana dashboards and Prometheus alerts required no updates)
    - [ ] The metrics names did change (Grafana dashboards and Prometheus alerts required an update)
- Logs checks:
    - [ ] The logs do not show any errors after the change
- PodSecurityPolicy checks:
    - [ ] Any changed Pod is covered by Kubernetes Pod Security Standards
    - [ ] Any changed Pod is covered by Gatekeeper Pod Security Policies
    - [ ] The change does not cause any Pods to be blocked by Pod Security Standards or Policies
- NetworkPolicy checks:
    - [x] Any changed Pod is covered by Network Policies
    - [ ] The change does not cause any dropped packets in the NetworkPolicy Dashboard
- Audit checks:
    - [ ] The change does not cause any unnecessary Kubernetes audit events
    - [ ] The change requires changes to Kubernetes audit policy
- Falco checks:
    - [ ] The change does not cause any alerts to be generated by Falco
- Bug checks:
    - [ ] The bug fix is covered by regression tests

Xartos reviewed (approved):
LGTM!

lunkan93 commented:
> I added a `tekton-monitoring` chart with the ServiceMonitor and "failed pipelines" alert. I set up the alert in the way it should work (the metric changed name from `tekton_pipelines_controller_pipelinerun_count` to `tekton_pipelines_controller_pipelinerun_total`, see [Tekton docs](https://tekton.dev/vault/pipelines-v1.0.x-lts/metrics/)), but I haven't actually gotten it to work because the metric doesn't appear for me. @lunkan93 Would you be able to test using this new chart and seeing if you have the same issue?

Can't promise that I'll have time to try it this week, but otherwise I can give it a go next week.

Ajarmar commented:
> This should be ready with a rebase and addressing some of the comments, since we already sorted out the metrics question, right?

Yes, the metrics problem should be solved. I pushed some changes now that should resolve the comments and some other things that I found myself.

anders-elastisys commented on migration/v0.51/apply/10-tekton-upgrade.sh:
```diff
@@ -0,0 +1,108 @@
+
+      kubectl_do sc label crd resolutionrequests.resolution.tekton.dev app.kubernetes.io/managed-by=Helm --overwrite
+      kubectl_do sc annotate crd resolutionrequests.resolution.tekton.dev meta.helm.sh/release-name=tekton-pipelines
+      kubectl_do sc annotate crd resolutionrequests.resolution.tekton.dev meta.helm.sh/release-namespace=tekton-pipelines
```
That is fine with me :P

anders-elastisys reviewed (changes requested):
I suggested a bad change, sorry, once my comment is fixed I will approve, as I think this looks good :)

Ajarmar commented on migration/v0.51/apply/80-apply.sh:
```diff
@@ -0,0 +1,57 @@
+)
+declare -a skipped_sc
+skipped_sc=(
+  app=tekton
```
Thanks for catching that! Updated it now.

Ajarmar commented on helmfile.d/values/tekton.gotmpl:
Yeah, we've only ran them with one replica, but it might be nice to upstream it anyway.
